app/controllers/projects_controller.rb

```diff
@@ -333,6 +333,46 @@ def create_reference
     end
   end
 
+  #Fang retuen the data for the progress of a project.
+  def progress
+    id = params[:id]
+    project = Project.find(id)
+    status = project.miga.datasets.map(&:status)
+    incomplete = status.count(:incomplete)
+    total = status.size
+    inactive = status.count(:inactive)
+    complete = (total - incomplete) - inactive
+    p_incomplete = (incomplete.to_f * 100.0)/total
+    p_complete = 100.0 - p_incomplete
+
+    p_complete = p_complete.round(2)
+   
+    #reading outputfile
+    f = File.expand_path("daemon/MiGA\:#{project.miga.name}.output", project.miga.path)
```

Comment:
There is no need to escape the colon, that should be handled below when accessing the file
```suggestion
    f = File.join(project.miga.path, 'daemon', "MiGA:#{project.miga.name}.output")
```